tgui/packages/tgui/interfaces/ZoldorfPlayerShop/type.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author Glowbold (https://github.com/pgmzeta)
+ * @license MIT
+ */
+
+import { BooleanLike } from "common/react";
+import { InfernoNode } from "inferno";
+
+export interface ZoldorfPlayerShopData {
+  soul_products: ZoldorfSoulProductData[],
+  credit_products: ZoldorfCreditProductData[],
+  credits: number,
+  user_soul: number,
+}
+
+export interface ZoldorfProductData {
+  name: string,
+  stock: number,
+  infinite: BooleanLike,
+  img: string,
+}
+
+export interface ZoldorfSoulProductData extends ZoldorfProductData {
+  soul_percentage: number,
+}
+
+export interface ZoldorfCreditProductData extends ZoldorfProductData {
+  price: number,
+}
+
+export interface ZoldorfProductListProps extends ZoldorfProductData {
+  children: InfernoNode,
+}
```

Comment:
Likely overkill, but with as written I'm assuming the idea here is for these to be mutually exclusive? If so, let me know and I'll write an example of how to use union types in Typescript to help with it.

Reply:
Yes please, I would like to learn how to use union types here

Reply:
Pester me in Discord (`@mordent`) if I don't do this in the next few days.